File: frontend/src/types/plans.ts
export type FeatureName =
  | "Alias record"
  | "Anycast DNS"
  | "Full API access"
  | "Multiple-layered DoS defense";

export interface Plan {
  name: string;
  price: number;
  period: string;
  users: number;
  description: string;
  features: Record<FeatureName, boolean>;
}

export interface PlansResponse {
  plans: Plan[];
  featureList: FeatureName[];
}

File: backend/app/schemas/plans.py
from pydantic import BaseModel
from typing import Dict, List

class Plan(BaseModel):
    name: str
    price: int
    period: str
    users: int
    description: str
    features: Dict[str, bool]

class PlansResponse(BaseModel):
    plans: List[Plan]
    featureList: List[str]
